src/main/scala/nl/knaw/dans/easy/stage/fileitem/EasyFileMetadata.scala

```diff
@@ -15,13 +15,18 @@
  */
 package nl.knaw.dans.easy.stage.fileitem
 
+import java.io.File
+
 import scala.util.Try
 
 object EasyFileMetadata {
   def apply(s: FileItemSettings): Try[String] = Try {
+      val parentPath = s.pathInDataset.get.getParentFile
```

Comment:
That's right, in essence this should return `Option[String]`. But since it's always used within the `Try` monadic context (i.e. in a for-comprehension of `Try`) you'll see yourself calling this like `efmd <- Try { EasyFileMetadata(s).get }`. It's a shame that `Option[T]` has no nice conversion method to `Try[T]`. What you really want is to be able to compose `Option` and `Try` instead of constantly converting between them.

Hint: @rvanheest knows about Monad Transformers ;)